### DISCUSSION TITLE
Is it possible to build static library of libOnnxruntime if somebody is aware? We are stuck. As per our business use case we can not use .so library.

### DISCUSSION THREAD

**Discussion by dsaneja** [microsoft/onnxruntime#16772] [General]
Is it possible to build static library of libOnnxruntime if somebody is aware? We are stuck. As per our business use case we can not use .so library.

Feeling really stuck on the same.

**Comment by RyanUnderhill** [comment]
It's possible for the core code, see the onnxruntime test library for an example, it statically links the files.

But the providers like cuda/etc are still setup to build/load as shared libraries. You'd have to do code changes to make this work, as it is unsupported.

**Reply by dsaneja** [reply]
Any specific command to build it statically if you are aware  please.

**Reply by RyanUnderhill** [reply]
The libraries are always built, you can see the test library statically linking here: https://github.com/microsoft/onnxruntime/blob/3252ff2cb7e9e29d400f4c3ed0fb123f2806e7df/cmake/onnxruntime_unittests.cmake#L562

You can also see someone else doing similar in this issue: https://github.com/microsoft/onnxruntime/issues/1472
And this issue: https://github.com/microsoft/onnxruntime/issues/8555

Sorry I couldn't find more accurate instructions, as it's recommended to use the shared library. I'd suggest trying to static link with some of the ones listed there and then add more as you see unresolved externals. Hopefully the issue thread gets you most of the way easily!
